contract-examples/contracts/ExampleFeeManager.sol

```diff
@@ -84,3 +103,131 @@ contract ExampleFeeManager is AllowList {
+
+    address exampleAddress = address(example);
+    address thisAddress = address(this);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    // TODO: make this a const (role: ADMIN = 2)
+    assertEq(manager.readAllowList(thisAddress), 2);
+    assertEq(manager.readAllowList(exampleAddress), 0);
+
+    manager.setEnabled(exampleAddress);
+
+    // TODO: make this a const (role: ENABLED = 1)
+    assertEq(manager.readAllowList(exampleAddress), 1);
+  }
+
+  function test_changeFees() public {
+    ExampleFeeManager example = new ExampleFeeManager();
+    address exampleAddress = address(example);
+
+    IFeeManager manager = IFeeManager(FEE_MANAGER_ADDRESS);
+
+    manager.setEnabled(exampleAddress);
+
+    FeeConfig memory config = example.getCurrentFeeConfig();
+
+    FeeConfig memory newFeeConfig = FeeConfig({
+      gasLimit: CCHAIN_GAS_LIMIT,
+      targetBlockRate: CCHAIN_TARGET_BLOCK_RATE,
+      minBaseFee: CCHAIN_MIN_BASE_FEE,
+      targetGas: CCHAIN_TARGET_GAS,
+      baseFeeChangeDenominator: CCHAIN_BASE_FEE_CHANGE_DENOMINATOR,
+      minBlockGasCost: CCHAIN_MIN_BLOCK_GAS_COST,
+      maxBlockGasCost: CCHAIN_MAX_BLOCK_GAS_COST,
+      blockGasCostStep: CCHAIN_BLOCK_GAS_COST_STEP
+    });
+
+    assertNotEq(config.gasLimit, newFeeConfig.gasLimit);
+    // target block rate is the same for wagmi and cchain
+    // assertNotEq(config.targetBlockRate, newFeeConfig.targetBlockRate);
+    assertNotEq(config.minBaseFee, newFeeConfig.minBaseFee);
+    assertNotEq(config.targetGas, newFeeConfig.targetGas);
+    assertNotEq(config.baseFeeChangeDenominator, newFeeConfig.baseFeeChangeDenominator);
+    // min block gas cost is the same for wagmi and cchain
+    // assertNotEq(config.minBlockGasCost, newFeeConfig.minBlockGasCost);
+    assertNotEq(config.maxBlockGasCost, newFeeConfig.maxBlockGasCost);
+    assertNotEq(config.blockGasCostStep, newFeeConfig.blockGasCostStep);
+
+    example.enableCChainFees();
+
+    FeeConfig memory changedFeeConfig = example.getCurrentFeeConfig();
+
+    assertEq(changedFeeConfig.gasLimit, newFeeConfig.gasLimit);
+    // target block rate is the same for wagmi and cchain
+    // assertEq(changedFeeConfig.targetBlockRate, newFeeConfig.targetBlockRate);
+    assertEq(changedFeeConfig.minBaseFee, newFeeConfig.minBaseFee);
+    assertEq(changedFeeConfig.targetGas, newFeeConfig.targetGas);
+    assertEq(changedFeeConfig.baseFeeChangeDenominator, newFeeConfig.baseFeeChangeDenominator);
+    // min block gas cost is the same for wagmi and cchain
+    // assertEq(changedFeeConfig.minBlockGasCost, newFeeConfig.minBlockGasCost);
+    assertEq(changedFeeConfig.maxBlockGasCost, newFeeConfig.maxBlockGasCost);
+    assertEq(changedFeeConfig.blockGasCostStep, newFeeConfig.blockGasCostStep);
+
+    assertEq(example.getFeeConfigLastChangedAt(), block.number);
+
+    example.enableCustomFees(config);
+  }
+
+  function test_minFeeTransaction() public {
+    // noop
```

Comment:
what is this function doing?

Reply:
https://github.com/ava-labs/subnet-evm/blob/3eec654c97d73dfc7a498e9ad1e19f5ad0bc1a1b/contract-examples/test/fee_manager.ts#L40

Testing that the transactions succeeds

Reply:
Can we add comments here that explains why it is empty?

Reply:
Then I have to add the same comment to every empty function and I suspect there will be many of these. I really don't want to optimize for being able to view this file without also looking at the accompanying `.ts` file.

Reply:
Maybe... I thought about that. I like having some kind of name when reading the test in TS though.

Reply:
> Then I have to add the same comment to every empty function and I suspect there will be many of these. I really don't want to optimize for being able to view this file without also looking at the accompanying .ts file.

I don't think this is a problem. Sounds very useful to anyone reading the code without context. eg,
```
// this function is empty as the test only requires the transaction to succeed. 
```

Reply:
done